brian2tools/plotting/base.py

```diff
@@ -151,3 +152,40 @@ def brian_plot(brian_obj,
     else:
         raise NotImplementedError('Do not know how to plot object of type '
                                   '%s' % type(brian_obj))
+
+
+def add_background_pattern(axes, hatch='xxx', fill=True, fc=(0.9, 0.9, 0.9),
+                           ec=(0.8, 0.8, 0.8), zorder=-10, **kwds):
+    '''
+    Add a "hatching" pattern to the background of the axes (can be useful to
+    make a difference between "no value" and a value mapped to a color value
+    that is identical to the background color). By default, it uses a cross
+    hatching pattern in gray which can be changed by providing the respective
+    arguments. All additional keyword arguments are passed on to the
+    `~matplotlib.patches.Rectangle` intializer.
+
+    Parameters
+    ----------
+    axes : `matplotlib.axes.Axes`
+        The axes where the
+    hatch : str, optional
+        See `matplotlib.patches.Patch.set_hatch`. Defaults to `'xxx'`.
+    fill : bool, optional
+        See `matplotlib.patches.Patch.set_fill`. Defaults to `True`.
+    fc : mpl color spec or None or 'none'
+        See `matplotlib.patches.Patch.set_facecolor`. Defaults to
+        `(0.9, 0.9, 0.9)`.
+    ec : mpl color spec or None or 'none'
+        See `matplotlib.patches.Patch.set_edgecolor`. Defaults to
+        `(0.8, 0.8, 0.8)`.
+    zorder : int
+        See `matplotlib.artist.Artist.set_zorder`. Defaults to `-10`.
+    '''
+    xmin, xmax = axes.get_xlim()
+    ymin, ymax = axes.get_ylim()
+    xy = (xmin,ymin)
```

Comment:
You caught me copy&pasting from stackoverflow ;) (in this case, I think the lines are trivial enough to not make a formal attribution necessary...).